web/src/scripts/containers/TabbedEditor.jsx

```diff
@@ -126,20 +222,61 @@ class TabbedEditor extends Component {
     })
   }
 
-  renderFlowInstallationToast() {
-    const linkStyle = {
-      textDecoration: 'underline',
+  onFocusTab = (tabId) => this.props.dispatch(openFile(this.props.filesByTabId[tabId].path))
+
+  onCloseTab = (tabId) => this.props.dispatch(closeTabWindow(tabId))
+
+  onCloseConfigErrorToast = () => this.props.dispatch(clearConfigError())
+
+  onCloseFlowErrorToast = () => this.props.dispatch(setFlowError(null))
+
+  onInstallFlow = () => {
+    const {rootPath, npmRegistry} = this.props
+
+    this.props.dispatch(setFlowError(null))
+    installAndStartFlow(rootPath, npmRegistry)
+  }
+
+  toggleConsole = () => {
+    const {consoleVisible} = this.props
+
+    this.props.dispatch(setConsoleVisibility(!consoleVisible))
+  }
+
+  togglePackager = (isRunning) => {
+    if (isRunning) {
+      this.props.dispatch(stopPackager())
+    } else {
+      this.props.dispatch(runPackager())
     }
+  }
+
+  saveScrollHeight = (scrollHeight) => this.props.dispatch(setConsoleScrollHeight(scrollHeight))
+
+  // Delay allows key events to finish first?
+  // TODO: move search menu to top level and take care of this on that refactor
+  onRequestCloseSearchMenu = () => {
+    setTimeout(() => {
+      this.refs.editor.getDecoratedComponentInstance().focus()
+    }, 200)
```

Comment:
This is some ancient code. I don't think delaying 0 works. CodeMirror internally delays 10 or 50 something to batch events.